### PR TITLE
[Fix](Planner) fix varchar does not show real length

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

Problem:
when we create table with datatype varchar(*), we regard it to be max length by default. But when we desc, it does not show 
real length but show varchar(*)
Reason:
when we upgrade version from 2.0.1 to 2.0.2, we support new feature of creating varchar(*), and it shows the same way with
ddl schema. So user would confuse of the length of varchar
Solved:
change the showing of varchar(*) to varchar(65533), which in compatible with hive

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/trino/LogicalPlanTrinoBuilder.java [hunk 1]
```diff
@@ -175,7 +176,8 @@ protected Expression visitCast(io.trino.sql.tree.Cast node, ParserContext contex
         Expression expr = visit(node.getExpression(), context, Expression.class);
         DataType dataType = mappingType(node.getType());
         Expression cast = new Cast(expr, dataType);
-        if (dataType.isStringLikeType() && ((CharacterType) dataType).getLen() >= 0) {
+        if (dataType.isStringLikeType() && ((CharacterType) dataType).getLen() >= 0
+                && !(dataType.isVarcharType() && ((VarcharType) dataType).isWildcardVarchar())) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/trino/LogicalPlanTrinoBuilder.java:180]: add comment to explain why max length not need to convert to `substr`

**Review by englefly — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/trino/LogicalPlanTrinoBuilder.java (hunk 1 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/parser/trino/LogicalPlanTrinoBuilder.java:181]: done

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Comment by LiBinfeng-01** [comment]
run buildall

**Pull request merged**